controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -435,6 +435,29 @@ func (r *DataScienceClusterReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
+func (r *DataScienceClusterReconciler) watchDataScienceClusterForDSCI(a client.Object) []reconcile.Request {
+	instanceList := &dsc.DataScienceClusterList{}
+	err := r.Client.List(context.TODO(), instanceList)
+	if err != nil {
+		return nil
+	}
+	var requestName string
+	switch {
+	case len(instanceList.Items) == 1:
+		requestName = instanceList.Items[0].Name
+	case len(instanceList.Items) == 0:
+		requestName = "default-dsc"
+	default:
+		return nil
+	}
+	// When DSCI CR gets created, trigger reconcile function
+	if a.GetObjectKind().GroupVersionKind().Kind == "DSCInitialization" || a.GetName() == "default-dsci" {
```

Comment:
just out of curiosity, why do you need to check for ```|| a.GetName() == "default-dsci"```? what is the scenario where that would trigger?

Reply:
I first used the check only on the` .Kind`, but i think in some cases it returns empty string, this also happens on `.Group`
my understanding it actually comes from https://github.com/kubernetes-sigs/controller-runtime/issues/1517
thats why we usually client.Object.GetName() in most of our code base. in theory, in ODH user can create their  DSCI CR with any name they want, in downstream, till we have webhook in place, default-dsci will the only valid name for DSCI CR.

Reply:
thank you. We might need to find an alternative to checking for the name directly, but good enough for now.